src/editor/classes/CodeMirrorInit.js

```diff
@@ -0,0 +1,59 @@
+"use strict";
+var editor;
+$(document).ready(function () {
+    editor = CodeMirror($(".editor")[0], {
+        lineNumbers: true,
+        mode: { name: "" },
+        highlightSelectionMatches: {style: "selectedHighlight", wordsOnly: true, delay: 25},
+        styleSelectedText: true,
+        styleActiveLine: true,
+        foldGutter: true,
+        gutters: ["CodeMirror-linenumbers", "CodeMirror-foldgutter"],
+        indentWithTabs: true,
+        indentUnit: 4,
+        tabSize: 4,
+        matchBrackets: true,
+        extraKeys: {"Ctrl-Space": "autocomplete"},
+        theme: _defaultTheme
+    });
+    
+    editor.addKeyMap({
+        "Tab": function (cm) {
+            if (cm.somethingSelected()) {
+                var sel = editor.getSelection("\n");
+                // Indent only if there are multiple lines selected, or if the selection spans a full line
+                if (sel.length > 0 && (sel.indexOf("\n") > -1 || sel.length === cm.getLine(cm.getCursor().line).length)) {
+                    cm.indentSelection("add");
+                    return;
+                }
+            }
+
+            if (cm.options.indentWithTabs)
+                cm.execCommand("insertTab");
+            else
+                cm.execCommand("insertSoftTab");
+        },
+        "Shift-Tab": function (cm) {
```

Comment:
So, button presses are pretty much events too. Might be a nice idea to move these two functions into EditorEventHook.js as well?